scheduler/clients/javascript/lib/calendarClient.ts

```diff
@@ -15,49 +15,142 @@ import type {
 import type { Timespan } from './eventClient'
 import { convertInstanceDates } from './helpers/datesConverters'
 
+/**
+ * Request for creating a calendar
+ */
 type CreateCalendarRequest = {
+  /**
+   * Timezone used in the calendar
+   */
   timezone: string
+  /**
+   * Start of the week used in the calendar
+   * @format 0-6 - 0 is Monday, 1 is Tuesday, etc.
+   * @default 0
+   */
   weekStart?: number
+  /**
+   * Possible metadata
+   */
   metadata?: Metadata
 }
 
+/**
+ * Request for updating a calendar
+ */
 type UpdateCalendarRequest = CreateCalendarRequest
 
+/**
+ * Response for getting the events of a calendar
+ */
 type GetCalendarEventsResponse = {
+  /**
+   * Calendar object
+   */
   calendar: Calendar
+  /**
+   * List of events with their instances
+   */
   events: {
+    /**
+     * Event object
+     */
     event: CalendarEvent
+    /**
+     * List of instances of the event
+     * Especially useful for recurring events
+     */
     instances: CalendarEventInstance[]
   }[]
 }
 
+/**
+ * Response for getting a calendar
+ */
 type CalendarResponse = {
+  /**
+   * Calendar object
+   */
   calendar: Calendar
 }
 
+/**
+ * Payload sent for enabling the sync of a calendar with an external calendar
+ */
 type SyncCalendarInput = {
+  /**
+   * Uuid of the user
+   */
   userId: string
```

Comment:
[suggestion] I like to use type aliases for common strings that have a canonical format.

You can then also comment on the type alias to provide an example format, so you don't need to do it everywhere.

e.g.

```ts
type UUID = string
...
calendarId: UUID
```

Especially in this PR, it's not clear to me if these IDs are UUIDs or not.